client/src/components/Add-Edit-Form/index.js

```diff
@@ -1,14 +1,108 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
-import { Form, Input, Button, message } from 'antd';
+import { Form, Input, Button, message, Spin } from 'antd';
 import './style.css';
 import AdminContainer from '../AdminContainer';
 
 const axios = require('axios');
 
-const AddEditForm = ({ formType, addLink, editLink, cohortId, inputData }) => {
-  const onFinish = async (values) => {
+const redirectFunc = (push, formType, cohortId, projectType) => {
+  if (formType === 'cohort') {
+    push('/admin/cohorts');
+  } else if (formType === 'student') {
+    push(`/admin/cohorts/${cohortId}/students`);
+  } else {
+    push(`/admin/cohorts/${cohortId}/projects?type=${projectType}`);
+  }
+};
+
+class AddEditForm extends Component {
+  state = {
+    myData: {},
+    // eslint-disable-next-line react/destructuring-assignment
+    addOrEdit: this.props.addLink,
```

Comment:
@MohammedAlghazali What is the purpose of addOrEdit? It looks like it's set to `edit` inside componentDidMount() after the network requests finish.  Is it intended to be used like a loading flag?

Reply:
hello
thank you for reviewing, I'm very pleased.
this component used as add form and edit form, I make addOrEdit because I need to make add form display directly when it makes render, but in the edit form, I get the data from the server so I want to display the form after the data back so I can display it in the form inputs value

Reply:
in case if the form works for edit, (this.props.addLink) will be undefined so the component will not render until the data back from server.

Reply:
Ah, okay I understand now.
Maybe, as an alternative, you could have an `isLoading` boolean in `this.state` instead of addOrEdit being used for this.  It would start as `false` and then inside componentDidMount(), you would only set it to true if you were in edit mode and starting axios requests.  Then you'd set it back to false when they finished.  That might be a little more clear to other developers as to why it exists.  But maybe that can just be a good idea for next time :)